### PR TITLE
Add Active and ProhibitLogin to API

### LINKED ISSUE #15571: API: include user activation and sign-in status
- Gitea version (or commit ref): 1.14.1
- Git version: 2.17.1
- Operating system: Ubuntu 18.04, Gitea Binary Download
  
  
  
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
- Log gist:

## Description

`active` and `prohibit_login` can both be set via `PATCH /admin/users/{username}`, but are not included in the output of `GET /admin/users` or `GET /users/{username}`. Imho those are important information when working with the user list and should be added.

### PR DESCRIPTION
**Pull request by KN4CK3R**
Fixes #15571

I used pointers to hide the new fields instead of returning `false` by default. Opinions?

### CODE REVIEW

File: modules/convert/user.go [hunk 1]
```diff
@@ -55,6 +55,8 @@ func toUser(user *models.User, signed, authed bool) *api.User {
 		result.IsAdmin = user.IsAdmin
 		result.LastLogin = user.LastLoginUnix.AsTime()
 		result.Language = user.Language
+		result.IsActive = &user.IsActive
```
Comment (6543) [inline modules/convert/user.go]: only admin & user itselve should know this info?

**Comment by 6543** [comment]
I would not use pointers, since for client it does not make a difference ...

File: modules/convert/user.go (hunk 1 above)
Comment (KN4CK3R) [reply modules/convert/user.go]: I thought that may be a sensitive information.

Comment (6543) [reply modules/convert/user.go]: Oh sorry you already do, didnt expand the diff

File: modules/structs/user.go [hunk 2]
```diff
@@ -33,6 +33,10 @@ type User struct {
 	Created time.Time `json:"created,omitempty"`
 	// Is user restricted
 	Restricted bool `json:"restricted"`
+	// Is user active
+	IsActive *bool `json:"active,omitempty"`
```
Comment (6543) [inline modules/structs/user.go]: Why ref?

Comment (lunny) [reply modules/structs/user.go]: Yes, just use `bool` if it will always have a value.

**Review by 6543 — APPROVED** [review]

File: modules/structs/user.go (hunk 2 above)
Comment (lunny) [reply modules/structs/user.go]: Sorry! I take my words back. For non-admin users, hide the field is better. A nil bool means `Don't know if it's active`. @6543

Comment (KN4CK3R) [reply modules/structs/user.go]: In this case all of these fields should be omitted:

https://github.com/go-gitea/gitea/blob/8f0539c235d47954d54dcf25c08297ddf7260970/modules/convert/user.go#L56-L61

`IsAdmin` would be a pointer then and `IsAdmin` and `Language` would get the `omitempty` flag. `LastLogin` already has the flag.

Comment (6543) [reply modules/structs/user.go]: well if it is always false it does not leak any info ... onto normal users

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**